game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProPurchaseOption.java

```diff
@@ -282,17 +281,13 @@ private double calculateSupportFactor(
     final List<Unit> units = new ArrayList<>(ownedLocalUnits);
     units.addAll(unitsToPlace);
     units.addAll(unitType.create(1, player, true));
-    final SupportCalculationResult supportCalculationResult =
-        AvailableSupportCalculator.getSupport(
-            units, data.getUnitTypeList().getSupportRules(), defense, true);
-
-    final Set<List<UnitSupportAttachment>> supportsAvailable =
-        supportCalculationResult.getSupportRules();
-    final IntegerMap<UnitSupportAttachment> supportLeft = supportCalculationResult.getSupportLeft();
+    final SupportCalculator availableSupports =
+        new SupportCalculator(units, data.getUnitTypeList().getSupportRules(), defense, true);
 
     double totalSupportFactor = 0;
     for (final UnitSupportAttachment usa : unitSupportAttachments) {
-      for (final List<UnitSupportAttachment> bonusType : supportsAvailable) {
+      for (final List<UnitSupportAttachment> bonusType :
+          availableSupports.getSupportRules().values()) {
```

Comment:
` availableSupports.getSupportRules().values()` looks like a violation of the law of delimiter.

What would you think of replacing the above with a method like:
` availableSupports.getUnitSupportAttachments()`